### PR TITLE
added colorbar label

### LINKED ISSUE #4906: Add label to colorbar in map.peek()
It would be nice to set the colorbar label from the unit of the map data (from the BUNIT FITS keyword), currently it is missing:

Screenshot 2021-01-25 at 14 40 05

### PR DESCRIPTION
**Pull request by jeffreypaul15**
### Description

Fixes #4906
Adds label to colorbar aligning it vertically.
This would change the figure so new hashes will have to be generated if what I've done is right.

### CODE REVIEW

**Comment by nabobalis** [comment]
How does it come out?

**Comment by jeffreypaul15** [comment]
> How does it come out?

image

Should I alter the size?

**Comment by nabobalis** [comment]
Can we not move the label? Maybe at the top or bottom of the bar?

It looks a bit odd sticking out at the side. 

Maybe others disagree.

**Comment by jeffreypaul15** [comment]
> Can we not move the label? Maybe at the top or bottom of the bar?
> 
> It looks a bit odd sticking out at the side.
> 
> Maybe others disagree.
I suppose moving it to the bottom and increasing the size would make it better?

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -2052,7 +2052,8 @@ def peek(self, draw_limb=False, draw_grid=False,
                 pad = 0.12  # Pad to compensate for ticks and axes labels
             else:
                 pad = 0.05  # Default value for vertical colorbar
-            figure.colorbar(im, pad=pad)
+            figure.colorbar(im, pad=pad).set_label(str(self.unit)
```
Comment (dstansby) [inline sunpy/map/mapbase.py:2055]: You should be able to pass `label=...` directly to the `colorbar()` call here.

**Comment by dstansby** [comment]
I agree it looks odd, if there's an easy way to put it at the top I think that would work well.

**Comment by jeffreypaul15** [comment]
> I agree it looks odd, if there's an easy way to put it at the top I think that would work well.

image

Does this look better?

**Comment by nabobalis** [comment]
Maybe the font should be same size as the tick labels?

File: changelog/4903.feature.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Add label for colorbar in ```map.peek```.
```
Comment (nabobalis) [inline changelog/4903.feature.rst:1]:
```suggestion
Colorbar in `sunpy.map.GenericMap.peek` now has a unit label.
```

**Comment by jeffreypaul15** [comment]
> Maybe the font should be same size as the tick labels?

Yeah, I thought so too. Was just getting that done

**Comment by jeffreypaul15** [comment]
> Maybe the font should be same size as the tick labels?

image

Yeah, I thought so too. I can't seem to find the actual size used though. ```size=12``` seems to best work with it.

**Comment by nabobalis** [comment]
I feel like it looks better at the bottom.

**Comment by jeffreypaul15** [comment]
> I feel like it looks better at the bottom.

image

Is this alright?

File: sunpy/map/mapbase.py (hunk 1 above)
Comment (jeffreypaul15) [reply sunpy/map/mapbase.py:2055]: Yeah but the positioning of the label requires the set_label attribute right?

**Comment by nabobalis** [comment]
Fine by me, lets see if others agree.

File: changelog/4903.feature.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Colorbar in `sunpy.map.GenericMap.peek` now has a unit label.
```
Comment (nabobalis) [inline changelog/4903.feature.rst:1]: The filename should be 4930, not 4903.

File: sunpy/map/mapbase.py [hunk 4]
```diff
@@ -2052,7 +2052,8 @@ def peek(self, draw_limb=False, draw_grid=False,
...
+                                                    if self.unit is not None else "", rotation=0, labelpad = -50, y=-0.02, size=12)
```
Comment (nabobalis) [inline sunpy/map/mapbase.py:2056]: I think doing the "str(self.unit)" section on a new line and passing the result into color bar would be neater.

File: sunpy/map/mapbase.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/map/mapbase.py:2055]: The docs say "label='" is a keyword arg to colorbar.

**Comment by dstansby** [comment]
Yep, I think this looks great, modulo @nabobalis comments above 👍

**Comment by nabobalis** [comment]
I suspect some figure tests will need updating due to the change as well.

File: changelog/4903.feature.rst (hunk 3 above)
Comment (jeffreypaul15) [reply changelog/4903.feature.rst:1]: Yes I shall make that change, my apologies.

File: sunpy/map/mapbase.py (hunk 1 above)
Comment (jeffreypaul15) [reply sunpy/map/mapbase.py:2055]: Yeah position of the label does require the ```set_label``` attribute

**Comment by jeffreypaul15** [comment]
> I suspect some figure tests will need updating due to the change as well.

Would the tests need updating? I thought only the hashes would have to be changed.

**Comment by nabobalis** [comment]
> > I suspect some figure tests will need updating due to the change as well.
> 
> Would the tests need updating? I thought only the hashes would have to be changed.

Just the hashes.

**Comment by jeffreypaul15** [comment]
I was unable to generate the hades using figure tests last time when I was
working on ```draw_quadrangle```. Any way you can generate the hashes?

**Comment by jeffreypaul15** [comment]
> > > I suspect some figure tests will need updating due to the change as well.
> > 
> > 
> > Would the tests need updating? I thought only the hashes would have to be changed.
> 
> Just the hashes.

I'm not too sure on how to generate the hashes as the ```tox -e figure``` tests do not pass locally for me. (Similar issue when I was working on ```draw_quadrangle```

**Comment by nabobalis** [comment]
> > > > I suspect some figure tests will need updating due to the change as well.
> > > 
> > > 
> > > Would the tests need updating? I thought only the hashes would have to be changed.
> > 
> > 
> > Just the hashes.
> 
> I'm not too sure on how to generate the hashes as the `tox -e figure` tests do not pass locally for me. (Similar issue when I was working on `draw_quadrangle`

The figure tests on circleci tell you the hashes before and after in the logs.

File: sunpy/map/mapbase.py [hunk 5]
```diff
@@ -2052,7 +2052,9 @@ def peek(self, draw_limb=False, draw_grid=False,
                 pad = 0.12  # Pad to compensate for ticks and axes labels
             else:
                 pad = 0.05  # Default value for vertical colorbar
-            figure.colorbar(im, pad=pad)
+            colorbar_label = str(self.unit) if self.unit is not None else ""
+            figure.colorbar(im, pad=pad).set_label(colorbar_label,
+                                                   rotation=0, labelpad=-50, y=-0.02, size=12)
```
Comment (nabobalis) [inline sunpy/map/mapbase.py:2057]: Does this need to be two lines?

Comment (jeffreypaul15) [reply sunpy/map/mapbase.py:2057]: PEP 8 gives me an issue if the files aren't in 2 separate lines so

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the PR @jeffreypaul15

**Comment by jeffreypaul15** [comment]
> Thanks for the PR @jeffreypaul15

Always fun contributing to Sunpy! :rocket:

**Pull request merged**